src/utilities/HYPRE_utilities.h

```diff
@@ -35,19 +35,34 @@ extern "C" {
 #if defined(HYPRE_BIGINT)
 typedef long long int HYPRE_BigInt;
 typedef long long int HYPRE_Int;
+
+#define HYPRE_BIG_INT_MAX INT64_MAX
```

Comment:
I'm not sure this `HYPRE_BIG_INT_MAX` define and the others like it belong in this external user header file.  I think this is something we only need internally for the binary read/write.  Also, I am not sure about explicitly associating 32 and 64 to Int and BigInt.  Maybe these should be removed and the one `if` statement in `IJMatrix.c` can be dealt with differently.

Reply:
Yes, I actually think this whole block belongs to `general.h`, which already contains a lot of typedefs for integer and floating-point types. Should we move this block to `general.h`?

Regarding the other question, I agree this needs to be changed